solutions/src/test/java/ru/sberbank/school/task02/ReverseCalculatorTest.java

```diff
@@ -0,0 +1,125 @@
+                BigDecimal.valueOf(10000), 20, Beneficiary.CLIENT);
+        BigDecimal offer = null;
+        if (optionalOffer.isPresent()) {
+            offer = optionalOffer.get();
+        }
+        Assertions.assertEquals(BigDecimal.valueOf(0.0025).setScale(10, RoundingMode.HALF_UP), offer);
+    }
+
+    @Test
+    public void convertReversedNullOperationThrowsNPEWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(NullPointerException.class);
+        exceptionRule.expectMessage("No operation provided");
+        calculator.convertReversed(null, Symbol.USD_RUB, BigDecimal.valueOf(1), Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedNullSymbolThrowsNPEWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(NullPointerException.class);
+        exceptionRule.expectMessage("No symbol provided");
+        calculator.convertReversed(ClientOperation.BUY, null, BigDecimal.valueOf(1), Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedNullAmountThrowsNPEWithSpecificMessage() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(NullPointerException.class);
+        exceptionRule.expectMessage("No amount provided");
+        calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, null, Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedZeroAmountThrowsIAE() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("Wrong amount");
+        calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, BigDecimal.ZERO, Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedNegativeAmountThrowsIAE() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("Wrong amount");
+        calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, BigDecimal.valueOf(-1), Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedNegativeDeltaThrowsIAE() {
+        ExtendedFxConversionService calculator = factory.getExtendedFxConversionService(quotesService);
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("Negative delta is mindless");
+        calculator.convertReversed(ClientOperation.BUY, Symbol.USD_RUB, BigDecimal.TEN, -1, Beneficiary.CLIENT);
+    }
+
+    @Test
+    public void convertReversedNullBeneficiaryThrowsNPEWithSpecificMessage() {
```

Comment:
Abbreviation in name &#39;convertReversedNullBeneficiaryThrowsNPEWithSpecificMessage&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1856867778>*